src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -56,12 +56,12 @@ private BoundExpression LowerDeclarationPattern(BoundDeclarationPattern pattern,
             if (pattern.IsVar)
             {
                 Debug.Assert(input.Type == pattern.LocalSymbol.Type);
-                var assignment = _factory.AssignmentExpression(_factory.Local(pattern.LocalSymbol), input);
+                var assignment = _factory.AssignmentExpression(pattern.VariableAccess, input);
```

Comment:
I'm documenting this invariant of this field. If we need to relax that invariant in the future, we can do so at that time.

---
In reply to: [77407635](https://github.com/dotnet/roslyn/pull/13535#discussion_r77407635) [](ancestors = 77407635,77397389,77386791)